### DISCUSSION TITLE
Code reuse and BSD license.

### DISCUSSION THREAD

**Discussion by xfractalino** [lwjglgamedev/vulkanbook#57] [General]
Hello, I am currently following this tutorial and have a question regarding my project. I am developing a library that incorporates a Vulkan renderer, and I'm utilizing code snippets from the tutorial as a basis for my implementation. In some cases, I heavily modify the code, while in others, I directly copy and paste (e.g., for Java beans classes or records). My intention is to license my library under the LGPL v2.1.

However, I am uncertain about how to appropriately license each individual file, especially those derived from the tutorial. Here are my specific concerns:

1) Files that contain solely my own work: These files would be licensed under LGPL v2.1.
2) How should modified files, derived from the tutorial, be licensed?
3) How should unmodified files, derived from the tutorial, be licensed? I believe they should retain the BSD header and original author's copyright.

I would greatly appreciate any guidance or clarification on the appropriate licensing approach for these different scenarios.

**Comment by lwjglgamedev** [comment]
Hello,

Not being a license expert, please feel free to use as w you wish. I think if you give proper credit to the book, in the way you prefer, it should be ok. Even more, if you think a better license model could be more useful (in the sense of giving more flexibility), I'm totally open to change it.

**Reply by xfractalino** [reply]
I've done some research, and it seems that including your copyright notice in the license header of each file that uses your code will meet the requirements of the BSD license. I'll make sure to establish a connection between your copyright notice and the original BSD license, which I'll copy to the root directory of my project.
I believe there's no need for you to change the license. Even if we were to consider a more permissive license like MIT, the same procedure would be necessary. Going for something even more permissive than MIT might compromise your copyright rights.
Let's stick with the current plan, as it strikes a good balance between respecting your rights and allowing me to use your code effectively.
Thank you, and keep up the good work! Your tutorial has been really useful.

**Reply by xfractalino** [reply]
@lwjglgamedev I'm sorry, there was a mistake in my research. If you license the project with a MIT license, I can re-license the code under my LGPLv2 while retaining your copyright notice, which would be much better for me while I still give proper credit to you. So, if you agree to change the license to MIT it would be very nice and I would be very grateful. If not, I will do the procedure I described in my last comment. Let me know. Thank you for your time.

**Comment by lwjglgamedev** [comment]
Hi, said and done. As I said my goal is to not limit redistribution. I hope it helps.

**Reply by xfractalino** [reply]
Thank you! I will let you know once I make my project public :)
